Removed unnecessary warning for 'outputs' parameter

joshuahansel opened the pull request:
Closes #25909

Also, cleaned up some grammar in the other warning in this method.

joshuahansel commented:
Just checking which tests fail first.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1854129/) on 112ac91 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25910/site)

This comment will be updated on new commits.

joshuahansel committed:
Removed unnecessary PP outputs warning

The warning in question was redundant and less useful than
a check already being made in
AdvancedOutput::initPostprocessorOrVectorPostprocessorLists
but the check was broken in some cases because of _advanced_execute_on
not being consistent.

In AdvancedOutput::init, checks were being performed before
_advanced_execute_on was made consistent with the input parameters,
so the calls to initExecutionTypes was moved up before these
checks, since it has no dependence on the preceding calls in init().

Closes #25909

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1856571/) on 540d1d8 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>191b9d</th>
      <th colspan=4>#25910 540d1d</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.63%</td>
      <td>85.64%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>92451</td>
      <td>92443</td>
      <td>-8</td>
      <td>5</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15509</td>
      <td>15507</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25910/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25910/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25910/coverage/xfem)

</details>



This comment will be updated on new commits.

lindsayad commented on framework/include/outputs/AdvancedOutput.h:
```diff
@@ -412,35 +409,19 @@ AdvancedOutput::initPostprocessorOrVectorPostprocessorLists(const std::string &
-        mooseWarning("A VectorPostprocessor utilizes the 'outputs' parameter; however, vector "
-                     "postprocessor output is disabled for the '",
+      {
+        const bool is_pp_type = (execute_data_name == "postprocessors") ? true : false;
```
```suggestion
        const bool is_pp_type = (execute_data_name == "postprocessors");
```

joshuahansel replied:
🤦

lindsayad commented on framework/src/outputs/AdvancedOutput.C:
```diff
@@ -150,6 +150,10 @@ AdvancedOutput::initialSetup()
 AdvancedOutput::init()
 {
+  // Initialize the execution flags
+  for (auto & it : _advanced_execute_on)
+    initExecutionTypes(it.first, it.second);
```
```suggestion
  for (auto & [name, input] : _advanced_execute_on)
    initExecutionTypes(name, input);
```

lindsayad commented on test/tests/outputs/format/tests:
```diff
@@ -109,8 +109,7 @@
     input = 'pps_screen_out_warn.i'
-    expect_out = "A Postprocessor utilizes the 'outputs' parameter; however, postprocessor output is "
-                 "disabled for the 'console' output object."
+    expect_out = "The post-processor 'avg_block' has requested to be output by the 'console' output, but post-processor output is disabled for this output object"
```
I think "that" would read better than "this"

joshuahansel committed:
Addressed review
